File: day_cung.py
def f(x):
    # Định nghĩa hàm f(x) của phương trình
    return x**4 - 2*x**2

def daycung(a, b, epsilon):
    # Tìm nghiệm gần đúng của phương trình
    x = a - (b - a) * f(a) / (f(b) - f(a))
    if f(x) * f(a) < 0:
        while abs(x - b) > epsilon:
            b = x
            x = a - (b - a) * f(a) / (f(b) - f(a))
    else:
        while abs(x - a) > epsilon:
            a = x
            x = a - (b - a) * f(a) / (f(b) - f(a))
    return x

print("Nghiệm gần đúng:", daycung(-1.8, -1.2, 0.02)) 

File: day_cung.cpp
#include <bits/stdc++.h>
using namespace std;
float f(float x){
	return pow(x,3)+3*pow(x,2)-3;
} 
float daycung(float a, float b){
	float exp = 0.001;
	float x; 
	do{
		x = a-(b-a)*f(a)/(f(b)-f(a));
		if(f(a)*f(x)<0)		b = x;
		else 	a = x;
	}while(abs(a-b)>exp);
	return x;
}
int main(){
	float a,b; //(-3;-2)
	cout<<"Nhap 1 khoang ly nghiem a,b: ";
	cin>>a>>b;
	cout<<"Nghiem gan dung x = "<<daycung(a,b);
}
